src/features/view-dao-assets/DaoAssetsTable/DaoAssetsTable.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useEffect, useMemo, useRef, useState } from 'react';
+import type { FC } from 'react';
+
+import { convertAsset } from './lib/helpers';
+import { useDaoAssets } from '../../../lib/hooks';
+
+import { TableControls } from '../../ui';
+import { DaoAssetsTableRow } from './DaoAssetsTableRow';
+import { DaoAssetsTableCard } from './DaoAssetsTableCard';
+import {
+	Body,
+	Grid,
+	Header,
+	HeaderGroup,
+	Table,
+	TableStatus,
+	TableStatusMessage,
+	View
+} from '@zero-tech/zui/components';
+import type { Column } from '@zero-tech/zui/components/AsyncTable';
+
+import styles from './DaoAssetsTable.module.scss';
+
+////////////////////
+// DaoAssetsTable //
+////////////////////
+
+export type DaoAssetTableDataItem = {
+	amount: string | number;
+	decimals?: number;
+	image: string;
+	key: string;
+	name: string;
+	subtext: string;
+	amountInUSD: string;
+};
+
+type DaoAssetsTableProps = {
+	zna?: string;
+};
+
+const TABLE_COLUMNS: Column[] = [
+	{ id: 'title', header: 'Asset', alignment: 'left' },
+	{ id: 'qty', header: 'Quantity', alignment: 'right' },
+	{ id: 'amount', header: 'Value (USD)', alignment: 'right' }
+];
+
+// @note: this value is being used in TableControls.module.scss - change in both places
+const GRID_WIDTH_TOGGLE = 450;
+
+export const DaoAssetsTable: FC<DaoAssetsTableProps> = ({ zna }) => {
+	const containerRef = useRef<HTMLDivElement>(null);
+	const [view, setView] = useState<View>(View.TABLE);
+
+	const { isLoading, isEmpty, tableData } = useDaoAssetsTableData(zna);
+
+	useEffect(() => {
+		const resizeObserver = new ResizeObserver(() => {
+			if (containerRef.current) {
+				if (containerRef.current.offsetWidth <= GRID_WIDTH_TOGGLE) {
+					setView(View.GRID);
+				}
+			}
+		});
+		resizeObserver.observe(containerRef.current);
+		return () => resizeObserver.disconnect();
+	}, [containerRef]);
+
+	return (
+		<div className={styles.DaoAssetsTable} ref={containerRef}>
```

Comment:
Like with the zApp-Staking and zApp-NFTS work where we've switched out the use of `AsyncTable`, ideally we would have a generic table component in zUI to pull in. This file will look very similar to the other table files - albeit with a few differences.